src/Mod/Part/App/FeaturePartFuse.cpp

```diff
@@ -102,12 +102,15 @@ App::DocumentObjectExecReturn *MultiFuse::execute()
     TopoShape compoundOfArguments;
 
     // if only one source shape, and it is a compound - fuse children of the compound
-    if (shapes.size() == 1) {
+    int naucks_sanity_remaining=1000000; // hard abort after 1 million loops - will trigger "not enough shape objects linked" error below if ever reached
+    while (shapes.size() == 1 && naucks_sanity_remaining-- > 0) {

```

Comment:
Though I like the tribute to my questionable sanity; we can clean up this code a bit to make it easier to read and follow.

```suggestion
    const int maxIterations = 1'000'000; // will trigger "not enough shape objects linked" error below if ever reached
    for (int i = 0; shapes.size() == 1 && i < maxIterations; ++i) {
```
* use c++14 digit separator for magic number. Makes the number easy to read and comment about the value unnecessary.
* rename variable so it represent the data. Side effect: making code dry, boring, and readable.
* using for loop that counts upwards to make. While counting downwards is a common optimization, the normal `for (int i = 0; ...)` encapsulates the intent better and makes it more easy to read. It also removes the need for an extra const for the magic number, see wwmayer's comment about the linter warning.